lessons/module-2/intro-to-destructuring.md

```diff
@@ -0,0 +1,186 @@
+    
+### The Answer   
+    
+Assigning an Array to a variable
+    
+<img class="medium" src="https://github.com/turingschool/front-end-curriculum/blob/dfec040a99f1673538180127f053d7753b3c9a8e/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-array-example1.png"/>   
+     
+<img class="medium" src="https://github.com/turingschool/front-end-curriculum/blob/f3b5d757b797c33fc08ff93c80c9122f875f2a3a/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-array-example2.png"/> 
+    
+</section>
+
+## Destructuring Arrays:
+#### Example:
+Return to your group and work on this example.
+    
+```js
+let bigArr = ["number", 4,5,6]
+let [ important, ...secondArr ] = bigArr;
+console.log(imporant, secondArr)
+
+```
+    
+## Destructuring objects:
+The right side of the statement contains the Javascript object that we want to split into variables; the left side contains a “pattern” for corresponding properties of the object. This “pattern” is usually a list of variable names.
+
+#### Example :
+    
+##### Assigning to exisitng variables 
+        
+```js
+    var user = {    // Object we want to destructure
+        firstname: 'Jon',
+        lastname: 'Doe',
+        dateofbirth: '1990'
+    };
+
+// Destructuring the object into our variables
+var { firstname, lastname, dateofbirth } = user;
+console.log( firstname, lastname, dateofbirth);
+
+```
+
+
+<section class="answer">
+### The Answer     
+<img class="medium" src="https://github.com/turingschool/front-end-curriculum/blob/dfec040a99f1673538180127f053d7753b3c9a8e/assets/images/lessons/intro-to-destructuring/intro-to-destructuring-object.png"/>
+
+</section>
+    
+
+    
+
+##### Assigning to new variable names:
+<section class="call to action">
+    
+```js
+    var user = {    // Object we want to destructure
+        firstname: 'Jon',
+        lastname: 'Doe',
+        dateofbirth: '1990'
+    };
+
+// Destructuring the object into variables with
+// different names than the object variables
+var { firstname: fn, lastname: ln, dateofbirth: dob } = user;
+console.log( fn, ln, dob);
+```
+</section>  
+
+<section class"answer>
+The above code destructures the object into variables with a different name than the object property:
+</section
+##### Assigning to a variable with default:
+
+
+We can also assign default values to variables whose keys may not exist in the object we want to destructure. This will prevent our variable from having an undefined value being assigned to it. The code below demonstrates this:
+
+```js
+    var user = {    // Object we want to destructure
+        firstname: 'Jon',
+        lastname: 'Doe',
+        dateofbirth: '1990'
+    };
+
+// Destructuring the object into variables without 
+// assigning default values 
+var { firstname, lastname, country } = user;
+console.log("Without setting default values")
+console.log( firstname, lastname, country);
+
+// Destructuring the object into variables by 
+// assigning default values 
```

Comment:
I like how you continue to build on the previous example and add more options that can be done.  Very cool!